### PR TITLE
Adjust rendering of the inline tag

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/48794

Moves us from looking like this:

image

to:

image

This also keeps our baselines aligned, making text not feel off balance from the surrounding lines:

image

--

Note: i would like the top spacing to be about 1px less.  However, i couldn't find any system that would make that work reliably without going off the rails elsewhere with some combination of font/zoom/text-size.  The current approach worked universally for as many cases as i tried (i.e. zoom-levels from 80% to 400%, and font sizes from 8pt to 24pt.)

@gundermanc if there are any ideas on what could be done here.

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]
The parentheses in the after screenshot is unrelated, right? Because if they're hiding in the code somewhere I can't see 'em 😄

File: src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTag.cs [hunk 1]
```diff
@@ -156,15 +162,11 @@ public async Task<IReadOnlyCollection<object>> CreateDescriptionAsync(Cancellati
...
                 Child = block,
                 CornerRadius = new CornerRadius(2),
-                Height = textView.LineHeight - (0.25 * textView.LineHeight),
-                HorizontalAlignment = HorizontalAlignment.Center,
-                Margin = new Thickness(left, top: -0.20 * textView.LineHeight, right, bottom: 0),
-                Padding = new Thickness(1),
-
-                // Need to set SnapsToDevicePixels and UseLayoutRounding to avoid unnecessary reformatting
-                SnapsToDevicePixels = textView.VisualElement.SnapsToDevicePixels,
-                UseLayoutRounding = textView.VisualElement.UseLayoutRounding,
```
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTag.cs:166]: ❔ Is this no longer the case?

**Comment by JoeRobich** [comment]
How nitpicky do we want to be here?
image

**Comment by CyrusNajmabadi** [comment]
> The parentheses in the after screenshot is unrelated, right? Because if they're hiding in the code somewhere I can't see 'em

Yeah, it's not an exact before/after.  i was playing around with lots of stuff.

File: src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTag.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTag.cs:166]: nope.  and it's actually problematic. the highlight rect doesn't do this, and it means we more often than not show up misaligned with it.

**Comment by CyrusNajmabadi** [comment]
>How nitpicky do we want to be here?

still much better than where we are currently (where we're off by full pixels).   :)

Note: i'm ok with an alternate approach here.  but it's a bit challenging here to get everything to work perfect here.

**Pull request merged**